### PR TITLE
Moo Ver. 2

### PR DESCRIPTION
**Pull request by Rfpnj**
Here you go after 4 hours of refactoring, Version 2 of moo. Now with more Moo for you :)

### CODE REVIEW

File: src/com/base/game/character/body/Arm.java [hunk 1]
```diff
@@ -184,6 +184,25 @@ public String setType(GameCharacter owner, ArmType type) {
...
+					UtilText.transformationContentSB.append(
+								" Within a matter of moments, a layer of [pc.armFullDescription] has quickly grown over them, and, looking down,"
+										+ " you see your hair growing over the backs of your new hands as tough, hoof-like nails push out in place of regular, human-like ones."
+								+ " Despite their appearance, you're relieved to discover that your hands have lost none of their dexterity."
+								+ " As the transformation comes to an end, you see that at your upper-biceps, your new hair smoothly transitions into the [pc.skin] that's covering the rest of your body.</br>"
+								+ "You're left with anthropomorphic, [style.boldCowMorph(cow-like arms and hands)], which are covered in [pc.armFullDescription]."
+							+ "</p>");
+				} else {
+					UtilText.transformationContentSB.append(
+								" Within a matter of moments, a layer of [pc.armFullDescription] has quickly grown over them, and, looking down,"
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Arm.java]: pc.armFullDescription

File: src/com/base/game/character/body/Arm.java [hunk 2]
```diff
@@ -184,6 +184,25 @@ public String setType(GameCharacter owner, ArmType type) {
...
+										+ " [npc.she] sees [npc.her] hair growing over the backs of [npc.her] new hands as tough, hoof-like nails push out in place of regular, human-like ones."
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Arm.java]: I am not sure about this, but this implies the previous hands were human or human-like - will that always be true? What about if the player has harpy arms?

File: src/com/base/game/character/body/Ass.java [hunk 3]
```diff
@@ -163,6 +163,19 @@ public String setType(GameCharacter owner, AssType type) {
...
+			case COW_MORPH:
+				if (owner.isPlayer()) {
+					UtilText.transformationContentSB.append(
+							"You now have a [style.boldCowMorph(bovine ass)], covered in [pc.assFullDescription].</br>"
+							+ "You have also been left with [style.boldCowMorph(a bovine)] [pc.assholeFullDescription]."
+							+ "</p>");
+				} else {
+					UtilText.transformationContentSB.append(
+							"[npc.She] now has a [style.boldCowMorph(feline ass)], covered in [npc.assFullDescription].</br>"
+							+ "[npc.She] has also been left with [style.boldCowMorph(a feline)] [npc.assholeFullDescription]."
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Ass.java]: "style.boldCowMorph(a feline)" - did you mean bovine

File: src/com/base/game/character/body/Ear.java [hunk 4]
```diff
@@ -161,6 +161,23 @@ public String setType(GameCharacter owner, EarType type) {
 							+ "</p>");
 				}
 				break;
+			case COW_MORPH:
+				if (owner.isPlayer()) {
+					UtilText.transformationContentSB.append(
+								" They quickly grow into long ovals, and shift to flop over on the side of your head vastly different than a normal pair of human ears would."
+								+ " A layer of [pc.earFullDescriptionColour] grows to cover them, and as the transformation finishes,  discover that you can easily twitch your new cow-like ears back and forth."
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Ear.java]: ",  discover" <-- missing "you"

File: src/com/base/game/character/body/Horn.java [hunk 5]
```diff
@@ -129,6 +129,36 @@ public String setType(GameCharacter owner, HornType type) {
...
+					UtilText.transformationContentSB.append(
+							" Hard nubs push out from your upper temples, and you gasp as you feel "+getDeterminer(owner)+" short, sleek horns pushing out to curve up the sides of your head."
+							+ "</br>"
+							+ "You now have "+getDeterminer(owner)+" [style.boldCowMorph(feminine-looking bovine horns)]."
+							+ "</p>");
+				} else {
+					UtilText.transformationContentSB.append(
+								" Hard nubs push out from [npc.her] upper temples, and [npc.she] gasps as [npc.she] feels "+getDeterminer(owner)+" short, sleek horns pushing out to curve up the sides of [npc.her] head."
+								+ "</br>"
+								+ "[npc.Name] now has "+getDeterminer(owner)+" [style.boldDemon(feminine-looking bovine horns)]."
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Horn.java]: style.boldDemon <-- did you mean cowMorph?

File: src/com/base/game/character/body/Penis.java [hunk 6]
```diff
@@ -325,6 +325,19 @@ public String setType(GameCharacter owner, PenisType type) {
...
+							"Letting out an involuntary moan, you feel your penis shifting into a new form, and you're hit by a wave of overwhelming arousal as your shaft grows wider and the head flattens down.</br>"
+							+ "You now have an [style.boldCowMorph(bovine penis)], covered in [pc.penisFullDescription].</br>"
+							+ "You have [style.boldCowMorph([pc.ballsCount]"+(owner.isInternalTesticles()?" internal,":"")+" bovine balls)], covered in [pc.ballsFullDescription(true)], which produce [pc.cumColour(true)] [style.boldCowMorph(bovine cum)].");
+				} else {
+					UtilText.transformationContentSB.append(
+							"Letting out an involuntary moan, [npc.name] feels [npc.her] penis shifting into a new form, and [npc.she]'s hit by a wave of overwhelming arousal as [npc.her] shaft grows wider and the head flattens down.</br>"
+							+ "[npc.She] now has an [style.boldCowMorph(bovine penis)], covered in [npc.penisFullDescription].</br>"
+							+ "[She] has [style.boldCowMorph([npc.ballsCount]"+(owner.isInternalTesticles()?" internal,":"")+" bovine balls)], covered in [npc.ballsFullDescription(true)], which produce [npc.cumColour(true)] [style.boldCowMorph(bovine cum)].");
```
Comment (Pimvgd) [inline src/com/base/game/character/body/Penis.java]: this "[She]" is missing the npc prefix but I don't know if that matters

File: src/com/base/game/character/body/types/AssType.java [hunk 7]
```diff
@@ -19,6 +19,8 @@
 	
 	DOG_MORPH(BodyCoveringType.CANINE_FUR, AnusType.DOG_MORPH, Race.DOG_MORPH),
 	
+	COW_MORPH(BodyCoveringType.BOVINE_FUR, AnusType.SQUIRREL_MORPH, Race.COW_MORPH),
```
Comment (Pimvgd) [inline src/com/base/game/character/body/types/AssType.java]: AnusType.SQUIRREL_MORPH ... did you mean COW_MORPH?

File: src/com/base/game/character/body/types/BodyCoveringType.java [hunk 8]
```diff
@@ -951,6 +999,21 @@
 			Colour.naturalIrisColours,
 			Colour.dyeIrisColours),
 
+	EYE_COW_MORPH(Race.COW_MORPH,
+			"a pair of",
+			true,
+			"eyes",
+			"eye",
+			"horse-like",
```
Comment (Pimvgd) [inline src/com/base/game/character/body/types/BodyCoveringType.java]: cow eyes are horse like?

File: src/com/base/game/character/npc/NPC.java [hunk 9]
```diff
@@ -5546,6 +5554,8 @@ public String getPenetrationDescription(boolean initialPenetration, PenetrationT
 								break;
 							case AVIAN:
 								break;
+							case BOVINE:
+								return "You let out [pc.a_moan+] as you feel [npc.name]'s COW-like cock push into your [pc.pussy+].";
```
Comment (Pimvgd) [inline src/com/base/game/character/npc/NPC.java]: COW -> cow?

File: src/com/base/game/combat/SpecialAttack.java [hunk 10]
```diff
@@ -997,6 +998,61 @@ public boolean isConditionsMet(GameCharacter owner) {
...
+		public String getDescription(GameCharacter owner) {
+			if (owner.isPlayer())
+				return "Your anthropomorphic cow-like horns can be used to deliver a powerful attack.";
+			else
+				return UtilText.genderParsing(owner, owner.getName("The") + "'s anthropomorphic cow-like horns can be used to deliver a powerful attack.");
+		}
+
+		@Override
+		public boolean isConditionsMet(GameCharacter owner) {
+			if (owner.getHornType() == HornType.BOVINE_MALE)
```
Comment (Pimvgd) [inline src/com/base/game/combat/SpecialAttack.java]: replace with "return owner.getHornType() == HornType.BOVINE_MALE;"

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 11]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an pastoral look to them. The shelves look like they haven been crafted from blocks of sod supported by pieces of wood."
+					+ " These books have the general feeling of being about the fields and it's people."
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: its people (it's = it is, thus "it is people" which is wrong)

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 12]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an urban look to them. The shelves look like they haven been crafted from concrete and iron."
+					+ "These books have the general feeling of being about the city and it's people."
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: its people

File: src/com/base/game/inventory/clothing/ClothingType.java [hunk 13]
```diff
@@ -1013,6 +1050,54 @@ public String unequipText(GameCharacter clothingOwner, GameCharacter clothingRem
 		}
 	},
 	
+	NECK_COWBELL_COLLAR("a",
+			false,
+			"cowbell collar",
+			"A heavy leather collar, of the type worn by cow.",
```
Comment (Pimvgd) [inline src/com/base/game/inventory/clothing/ClothingType.java]: .... "worn by cows", maybe

File: src/com/base/game/inventory/clothing/ClothingType.java [hunk 14]
```diff
@@ -1013,6 +1050,54 @@ public String unequipText(GameCharacter clothingOwner, GameCharacter clothingRem
...
+			null,
+			
+			Colour.allMetalColours) {
+		@Override
+		public String equipText(GameCharacter clothingOwner, GameCharacter clothingRemover, boolean rough, AbstractClothing clothing, boolean applyEffects) {
+			if (clothingOwner.isPlayer() && clothingRemover.isPlayer())
+				return "You put on the cowbell collar, fastening the clasp at the back of your neck.";
+			else if (!clothingOwner.isPlayer() && !clothingRemover.isPlayer())
+				return UtilText.genderParsing(clothingOwner,
+						Util.capitaliseSentence(clothingOwner.getName("the")) + " puts a " + clothing.getName(true) + " around <her> neck, reaching around" + " to fasten the clasp at the back.");
```
Comment (Pimvgd) [inline src/com/base/game/inventory/clothing/ClothingType.java]: no need to concatenate two string literals (e.g. "a" + "b")  here, you can just turn them into one string literal (e.g. "ab")

**Comment by Pimvgd** [comment]
Also of note, the cow book is currently not enabled

File: src/com/base/game/character/body/Arm.java (hunk 1 above)
Comment (Rfpnj) [reply src/com/base/game/character/body/Arm.java]: It's Innoxia's code across all arms

File: src/com/base/game/character/body/Penis.java (hunk 6 above)
Comment (Rfpnj) [reply src/com/base/game/character/body/Penis.java]: It shouldn't matter

File: src/com/base/game/combat/SpecialAttack.java (hunk 10 above)
Comment (Rfpnj) [reply src/com/base/game/combat/SpecialAttack.java]: Not completely sure about this one, as it matches the style of the rest

**Comment by Rfpnj** [comment]
OK went through and fixed most things. A few I did not but gave the reason why above

File: src/com/base/game/character/body/Arm.java (hunk 1 above)
Comment (Pimvgd) [reply src/com/base/game/character/body/Arm.java]: @Innoxia then this is a bug that NEEDS fixing

Comment (Pimvgd) [reply src/com/base/game/character/body/Arm.java]: Only for horses is it bugged

Comment (Innoxia) [reply src/com/base/game/character/body/Arm.java]: I think I fixed that :s

File: src/com/base/game/character/body/Penis.java (hunk 6 above)
Comment (Innoxia) [reply src/com/base/game/character/body/Penis.java]: It won't parse without the 'npc' tag, so it should be [npc.She] (I'll fix this now :3)

**Pull request merged**